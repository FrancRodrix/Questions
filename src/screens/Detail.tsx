import { useNavigation } from "@react-navigation/core";
import React from "react";
import {
  View,
  Text,
  TouchableOpacity,
  SafeAreaView,
  StyleSheet,
} from "react-native";

interface Props {
  navigation;
}

const Detail = (props: Props) => {
  const Item = props.route.params.data;

  return (
    <SafeAreaView style={styles.mainContainer}>
      <TouchableOpacity
        style={styles.back}
        onPress={() => {
          navigation.goBack();
        }}
      >
        <Text>Back</Text>
      </TouchableOpacity>
      <View style={styles.box}>
        <Text style={{ textAlign: "center" }}>{Item.name}</Text>
        <Text style={{ textAlign: "center" }}>{Item.name_limited}</Text>
        <Text style={{ textAlign: "center" }}>{Item.designation}</Text>
      </View>
    </SafeAreaView>
  );
};

export default Detail;

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#fe4",
  },
  back: {
    position: "absolute",
    top: 60,
    left: 30,
  },
  box: {
    alignSelf: "center",
    borderWidth: 1,
    borderRadius: 10,
    padding: 20,

    backgroundColor: "#fef",
  },
});
