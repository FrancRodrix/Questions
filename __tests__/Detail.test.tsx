import "react-native";
import React from "react";
import Detail from "../src/screens/Detail";
import renderer from "react-test-renderer";

const navigation = {
  navigation: jest.fn(),
  params: {
    data: {
      name: "Delhi",
      name_limited: "Contry",
      designation: "asteriod",
    },
  },
};
// configure({ adapter: new Adapter() });
const tree = renderer.create(<Detail route={navigation} />);
test("renders correctly", () => {
  expect(tree).toMatchSnapshot();
});
