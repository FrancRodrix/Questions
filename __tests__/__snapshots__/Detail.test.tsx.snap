// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fe4",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 30,
        "opacity": 1,
        "position": "absolute",
        "top": 60,
      }
    }
  >
    <Text>
      GO BACK
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fef",
        "borderRadius": 10,
        "borderWidth": 1,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      xys
    </Text>
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      cvb
    </Text>
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      asd
    </Text>
  </View>
</RCTSafeAreaView>
`;
